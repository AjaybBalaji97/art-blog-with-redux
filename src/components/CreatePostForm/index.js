import React, { Component } from 'react';
import './CreatePostForm.css';


class CreatePostForm extends Component {
  render() {
    return (
        <div className="PostForm">
        <h1>New Blog Post</h1>
        <form onSubmit={this.handleSubmit}>
            <p> <label for="title">TITLE:</label><textarea id="title" type="text"  /></p>
            <p><label for="desc">DESCRIPTION</label><textarea id="desc" type="text"  /></p>
            <p><label for="date">Date</label><input id="date" type="date" /></p>
            <p><label for="image">IMAGE</label><input id="image" type="file" /></p>
            <p><input type="submit" /></p>
        </form>
            

        </div>
    );
  }
}

export default CreatePostForm;