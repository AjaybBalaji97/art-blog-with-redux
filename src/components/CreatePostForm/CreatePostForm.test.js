/**
 * @jest-environment node
 */
import React from 'react';
import renderer from 'react-test-renderer';
import CreatePostForm from '.';
import {shallow,mount} from 'enzyme';
import axios from 'axios';


describe('CreatePostForm',()=>{
    let postMock;
    beforeAll(()=>{
        postMock = jest.spyOn(axios,'post');
        postMock.mockImplementation(()=>Promise.resolve({data:'Posted Data'}))
    })
    afterAll(()=>{
        postMock.mockRestore()
    })
    it('should render CreatePostForm ',()=>{
        const tree=renderer.create(<CreatePostForm />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should call "handleDataChange" on changing any form field',()=>{
        const component= shallow(<CreatePostForm />) ;
        const mockFn=jest.fn();
        component.find("#title").simulate('change',{target:{id:'title',value:'title'}});
        expect(component.find("#title").length).toEqual(1);
        expect(component.state('title')).toEqual('title');
        //component.unmount();
     })
     it('should call "handlePostData" on clicking submit button',()=>{
        const component= shallow(<CreatePostForm />) ;
        //console.log(component.find('.submit'));
        const mockFn=jest.fn();
        component.instance().handlePostData=mockFn();
        component.find(".submit").simulate('click',{target:{value:'submit',preventDefault:jest.fn()}});
        expect(mockFn).toHaveBeenCalled();
        //component.unmount();
        jest.restoreAllMocks();
     })
     xit('should call "handlePostData" on clicking submit button',async ()=>{
        const component= shallow(<CreatePostForm />) ;
        //console.log(component.find('.submit'));
        const mockFn=jest.fn();
        //component.instance().handlePostData=mockFn();
        component.find(".submit").simulate('click',{target:{value:'submit',preventDefault:jest.fn()}});
        await component.instance().handlePostData()
        
        expect(component.instance().state.status).toEqual('');
        //component.unmount();
        jest.restoreAllMocks();
     })


});