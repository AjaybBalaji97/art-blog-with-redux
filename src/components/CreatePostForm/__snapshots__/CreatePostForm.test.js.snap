// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePostForm should render CreatePostForm  1`] = `
<div
  className="PostForm"
>
  <h1>
    New Blog Post
  </h1>
  <form>
    <p>
       
      <label>
        TITLE:
      </label>
      <textarea
        id="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </p>
    <p>
       
      <label>
        DESCRIPTION
      </label>
      <textarea
        id="description"
        onChange={[Function]}
        type="text"
        value=""
      />
    </p>
    <p>
       
      <label>
        Date
      </label>
      <input
        id="date"
        onChange={[Function]}
        type="date"
        value=""
      />
    </p>
    <p>
       
      <label>
        IMAGE
      </label>
      <input
        id="imageFile"
        onChange={[Function]}
        type="file"
        value=""
      />
    </p>
    <p>
      <button
        className="submit"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;
